### PR TITLE
feat: Select your agent component

### PR DESCRIPTION
**Pull request by mohandast52**
## Proposed changes

- Added the `<SetupYourAgent />` dummy component (pending integration).

Screenshot 2024-12-02 at 2 49 26 PM

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/SetupPage/SelectYourAgent.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import { usePageState } from '@/hooks/usePageState';
+import { useServices } from '@/hooks/useServices';
+import { AgentConfig } from '@/types/Agent';
+
+import { CardFlex } from '../styled/CardFlex';
+import { SetupCreateHeader } from './Create/SetupCreateHeader';
+
+const { Title, Text } = Typography;
+
+type EachAgentProps = { agentType: AgentType; agentConfig: AgentConfig };
```
Comment (Tanya-atatakai) [inline frontend/components/SetupPage/SelectYourAgent.tsx:20]:
```suggestion
type EachAgentProps = { agentType: AgentType; agentConfig: AgentConfig };

```

File: frontend/components/SetupPage/SelectYourAgent.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const EachAgent = ({ agentType, agentConfig }: EachAgentProps) => {
+  const { selectedAgentType, updateAgentType } = useServices();
+  const { goto } = usePageState();
+
+  const isCurrentAgent = selectedAgentType === agentType;
+
+  const handleSelectAgent = useCallback(() => {
+    updateAgentType(agentType);
+    goto(Pages.SetupYourAgent); // TODO: page to be added
```
Comment (Tanya-atatakai) [inline frontend/components/SetupPage/SelectYourAgent.tsx:29]: can we support dynamic page here? can be either onboarding part or main, but should somehow pass it

File: electron/public/agent-memeooorr-icon.png
Comment (Tanya-atatakai) [inline electron/public/agent-memeooorr-icon.png:1]: why should we have two sets of images? one here electron/public and another one in frontend/public?

Comment (truemiller) [reply electron/public/agent-memeooorr-icon.png:1]: images end up here after build, can probably be git ignored though as they'll be copied during release process

**Comment by truemiller** [comment]
these buttons should probably always show, be clickable

if user lands on this page, but decides not to switch, they must switch to the other agent, then repeat the process to get back

image

**Comment by truemiller** [comment]
> > if user lands on this page, but decides not to switch
> 
> IMO, the back button should handle that. We can pass the back navigation page as a prop. If something is already selected, it should navigate to the “Main” page; otherwise, it should go to the “Backup wallet” page. WDYT?

sure
would prefer enabling both options to not block user from their intention, but is minor

on props, i'd pass both `back` and `next` as this page will be used in 2 places

**Review by truemiller — APPROVED** [review]

**Review by Tanya-atatakai — APPROVED** [review]

**Pull request merged**